src/tests/ftest/checksum/basic_checksum.py

```diff
@@ -0,0 +1,178 @@
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import os
+import ctypes
+import time
+import agent_utils
+import server_utils
+import write_host_file
+
+from pydaos.raw import (DaosPool, DaosContainer, IORequest,
+                        DaosObj)
+from apricot import TestWithServers
+
+
+class ChecksumContainerValidation(TestWithServers):
+    """
+    Test Class Description:
+
+    :avocado: recursive
+    """
+    # pylint: disable=too-many-instance-attributes
+    def setUp(self):
+        super(ChecksumContainerValidation, self).setUp()
+        self.agent_sessions = None
+        self.pool = None
+        self.container = None
+        self.obj = None
+        self.ioreq = None
+        self.hostlist = None
+        self.hostfile = None
+        self.no_of_dkeys = None
+        self.no_of_akeys = None
+        self.array_size = None
+        self.record_length = None
+        server_group = self.params.get("name",
+                                       '/server_config/',
+                                       'daos_server')
+        self.hostlist = self.params.get("test_servers", '/run/hosts/*')
+        self.hostfile = write_host_file.write_host_file(self.hostlist,
+                                                        self.workdir)
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')[0]
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')[0]
+        self.record_length = self.params.get("length", '/run/record/*')
+
+        self.agent_sessions = agent_utils.run_agent(
+            self, self.hostlist)
+        server_utils.run_server(self, self.hostfile, server_group)
+
+        self.pool = DaosPool(self.context)
+        self.pool.create(self.params.get("mode", '/run/pool/createmode/*'),
+                         os.geteuid(),
+                         os.getegid(),
+                         self.params.get("size", '/run/pool/createsize/*'),
+                         self.params.get("setname", '/run/pool/createset/*'),
+                         None)
+        self.pool.connect(2)
+        self.csum = self.params.get("enable_checksum", '/run/container/*')
+        self.container = DaosContainer(self.context)
```

Comment:
Same as above. Please use TestContainer/TestContainerData in test_utils whichever suits your read/write needs better. Also It will be nice to reflect your container prop changes to the TestContainer.create() as well. It will be nice so that it can be used by other tests as well.

Reply:
TestContainer changes will be done as part of DAOS-3774. I don't want to land too many changes as part of this PR.